consensus/src/experimental/buffer_item.rs

```diff
@@ -172,4 +198,46 @@ impl BufferItem {
             Self::Aggregated(aggregated) => aggregated.aggregated_proof.ledger_info().commit_info(),
         }
     }
+
+    pub fn add_signature_if_matched(
+        &mut self,
+        target_commit_info: &BlockInfo,
+        author: Author,
+        sig: Ed25519Signature,
+    ) -> bool {
+        match self {
+            Self::Ordered(ordered) => {
+                if ordered
+                    .ordered_proof
+                    .commit_info()
+                    .match_ordered_only(target_commit_info)
+                {
+                    ordered.pending_votes.insert(author, sig);
```

Comment:
worth adding comments how do we handle the different executed votes added here

Reply:
added comments